images/pytorch-notebook/cuda/Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+ARG REGISTRY=quay.io
+ARG OWNER=jupyter
+ARG BASE_CONTAINER=$REGISTRY/$OWNER/scipy-notebook
+FROM $BASE_CONTAINER
+
+LABEL maintainer="Jupyter Project <jupyter@googlegroups.com>"
+
+# Fix: https://github.com/hadolint/hadolint/wiki/DL4006
+# Fix: https://github.com/koalaman/shellcheck/wiki/SC3014
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# Install PyTorch with pip (https://pytorch.org/get-started/locally/)
+# hadolint ignore=DL3013
+RUN pip install --no-cache-dir --extra-index-url=https://pypi.nvidia.com \
```

Comment:
One more thought - let's call this image cuda12 and I think we can add something like `--index-url 'https://download.pytorch.org/whl/cu121'` to the installation command.

When cuda13 will be released in the future, we won't break anyone by silently updating their image from cuda12 to cuda13. We will just add a new prefix tag `cuda13` and everyone who can and wants to upgrade will be able to do that with a small change.

Reply:
I added `--index-url 'https://download.pytorch.org/whl/cu121'` and renamed the `cuda` variant to `cuda12`.